### ISSUE TITLE
Custom Headers on HTTP webhook

### ISSUE THREAD

**Issue by jeremyjjbrown** [grafana/grafana#9813]
Allow Custom Headers to be set on an HTTP webhook.

![2017-11-07-123714_2421x1612_scrot](https://user-images.githubusercontent.com/1660010/32511869-c221dda4-c3ba-11e7-830e-83a8300d92ad.png)

Only allows Basic auth. Obviously this is very limiting. It would be nice to be able to set any Headers applicable. For example, auth with systems that want a Bearer Token could be supported without specifc plugins by including a "Authorization: Bearer $TOKEN".

This would also be useful for countless other applications.
 
I might be willing to implement this if the project owners agree that it should be added.

**Comment by TimonDDD** [comment]
Can there be something already realized or exists another way to authorized in AWS?

**Comment by jeremyjjbrown** [comment]
@TimonDDD  try a lambda with api gateway and set the webhook url to that.

https://docs.aws.amazon.com/lambda/latest/dg/with-on-demand-https-example-configure-event-source_1.html

**Comment by jeremyjjbrown** [comment]
The fact that Graphana owners do not even provide feedback after 7 months that this would or would not be worth implementing is disapointing.

**Comment by victor-leee** [comment]
A lot of applications require extra authentication from HTTP Header other than Authentication. Plz consider enroll this feature in next release maybe?

**Comment by santru** [comment]
Is there any resolution for this problem? Any way to get around it?

**Comment by shikha-gaur** [comment]
It's available now.

**Comment by tushar2013** [comment]
@shikha-gaur any link or example?

**Comment by hotpotato0** [comment]
@shikha-gaur  What version is available?

**Comment by jeremyjjbrown** [comment]
5 years after I offered to do this for free it's still unresolved. Sad.

**Comment by bruno-kakele** [comment]
+1

**Comment by zfrhv** [comment]
+1

**Comment by Meza1923** [comment]
+1

**Comment by utshau** [comment]
<img width="713" alt="Screenshot 2023-10-30 at 2 52 23 PM" src="https://github.com/grafana/grafana/assets/37466736/e7c58c4d-667b-4086-8488-012dcb7d7283">
There is authentication now, version I am seeing is v10

**Comment by utshau** [comment]
<img width="1218" alt="Screenshot 2023-10-30 at 2 57 21 PM" src="https://github.com/grafana/grafana/assets/37466736/1d1902db-098b-4770-83f6-55d164314eeb">
Now we have auth, version I am seeing is v10 but I am not sure from which version it started.

**Comment by gelicia** [comment]
This looks like it was implemented. Closing. If the above doesnt cover your use case, please comment in the ticket and I can figure out if there is a bug or we should reopen this feature request. Thank you!

**Comment by saketbairoliya2** [comment]
This doesn't seems to be resolved yet, I'm still not able to pass the custom headers to contact point. Want to also check, how can we do this from API?

**Comment by gelicia** [comment]
@saketbairoliya2 Here's the documentation on how to include the settings via provisioned yaml, the settings json in the API call should be the same, just in JSON https://grafana.com/docs/grafana/latest/alerting/set-up/provision-alerting-resources/file-provisioning/#webhook 

Let me know if that doesn't work for ya!

**Comment by saketbairoliya2** [comment]
Thanks @gelicia - But my use-case is to pass some arbitrary map of k:v pairs in the header. Is that supported? 

Example header k:v pairs that I want to add to the header: 
X-User-id: test-user
X-tenant-id: test-tenant

**Comment by gelicia** [comment]
Nope, I will reopen this and edit my comments that it is only a few authorization headers that were added since the ticket was made.

**Comment by gillesdemey** [comment]
See https://github.com/grafana/alerting/issues/134

**Comment by TrevorSquillario** [comment]
Related https://github.com/grafana/grafana/issues/50558

**Comment by JacobsonMT** [comment]
Fixed with: https://github.com/grafana/grafana/pull/103818
